src/volatile.rs

```diff
@@ -3,6 +3,12 @@
 #[repr(transparent)]
 pub struct ReadOnly<T: Copy>(T);
 
+impl<T: Debug + Copy> Debug for ReadOnly<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
```

Comment:
No need for the explicit lifetime `<'_>` here and in the other implementation below.